graviti/dataframe/column/series.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""The implementation of the Lazy Series."""
+
+
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Union, overload
+
+from graviti.dataframe.series import SeriesBase
+
+
+class Series(SeriesBase[int]):
+    """One-dimensional array.
+
+    Arguments:
+        data: The data that needs to be stored in Series. Could be ndarray or Iterable.
+        schema: Data type to force. Only a single dtype is allowed. If None, will be
+            inferred from `data`.
+        name: The name to the Series.
+        index: Index of the data, must have the same length as `data`.
+
+    Examples:
+        Constructing Series from a list.
+
+        >>> d = [1,2,3,4]
+        >>> series = Series(data=d)
+        >>> series
+        0 1
+        1 2
+        2 3
+        3 4
+        schema: int
+
+    """
+
+    _indices_data: Optional[Dict[int, int]]
+    _indices: Optional[List[int]]
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        data: Sequence[Any],
+        schema: Any = None,
+        name: Optional[Union[str, int]] = None,
+        index: Optional[List[int]] = None,
+    ) -> None:
+        if schema is not None:
+            # TODO: missing schema processing
+            pass
+
+        self._data = data
+        self.name = name
+        self._indices_data = (
+            index if index is None else {value: key for key, value in enumerate(index)}
+        )
+        self._indices = index
+
+    # @overload
+    # def __getitem__(self, key: slice) -> "Series":
+    #    ...
+
+    @overload
+    def __getitem__(self, key: int) -> Any:
+        ...
+
+    @overload
+    def __getitem__(self, key: Iterable[int]) -> "Series":
+        ...
+
+    def __getitem__(self, key: Union[int, Iterable[int]]) -> Any:
+        integer_location = self._get_loc_by_index(key)
+
+        if isinstance(integer_location, list):
+            new_data = [self._data[loc] for loc in integer_location]
+            return Series(new_data, name=self.name, index=integer_location)
+
+        return self._data[integer_location]
+
+    def _get_loc_by_index(self, key: Union[int, Iterable[int]]) -> Union[int, List[int]]:
```

Comment:
shall we add overload for `_get_loc_by_index`?

Reply:
What is the difference between `loc` and `location`? why we have `loc` and `location` in function name?